### PR TITLE
Async `Frame`

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

* There are a lot of commits. Luckily, each commit is small and corresponds to a method.
* The PR removes panics from a few methods and turns them into async. I don't want to split the PR because of a few panicking methods. But I used different commits for revertability.

## Why?

#1302

## Checklist

- [x] I have performed a self-review of my code
- [ ] I have added tests for my changes
- [ ] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by inancgumus** [comment]
Ah, mistakenly merged #1369 into this one 🤦

**Review by ankur22 — APPROVED** [review]
🎉 🚀 Let's GO!

File: browser/frame_mapping.go [hunk 1]
```diff
@@ -79,7 +79,11 @@ func mapFrame(vu moduleVU, f *common.Frame) mapping { //nolint:gocognit,cyclop
 				return nil, f.Fill(selector, value, opts) //nolint:wrapcheck
 			})
 		},
-		"focus": f.Focus,
+		"focus": func(selector string, opts goja.Value) *goja.Promise {
+			return k6ext.Promise(vu.Context(), func() (any, error) {
+				return nil, f.Focus(selector, opts) //nolint:wrapcheck
+			})
+		},
 		"frameElement": func() (mapping, error) {
```
[the remark this answers is not in the document]
Comment (inancgumus) [reply browser/frame_mapping.go:86]: Thanks for catching this 🙇 It seems it doesn't matter how many times I check these methods 🥲

[the remark this answers is not in the document]
Comment (inancgumus) [reply browser/frame_mapping.go]: Fixed here https://github.com/grafana/xk6-browser/pull/1367/commits/30073cf2390e2f9c2ddb2526ef63510364980942

**Review by ankur22 — APPROVED** [review]
LGTM 🚀

**Pull request merged**